File: Group 3/MessagingSystem.Api/Models/EnqueueRequest.cs
namespace WebApplication1.Models;

public class EnqueueRequest
{
    public required string DestinationUrl { get; set; }
    public Dictionary<string, string>? Headers { get; set; }
    public object? Payload { get; set; }
    
    public string? ClientMessageId { get; set; }
    public string? TenantId { get; set; }
    public string? CallbackUrl { get; set; }
    public DateTime? DeliverAtUtc { get; set; }
}

File: Group 3/mac/queue/queue/models/EnqueueRequest.swift
//
//  request.swift
//  queue
//
//  Created by Pablo Rodriguez on 26/2/26.
//
import Foundation

struct EnqueueRequest: Codable {
    var destinationUrl: String
    var headers: [String: String]?
    var payload: String?
    var clientMessageId: String?
    var tenantId: String?
    var callbackUrl: String?
    var deliverAtUtc: Date?
}
